server/src/main/java/com/hedera/block/server/verification/StreamVerificationHandlerImpl.java

```diff
@@ -0,0 +1,130 @@
+import com.hedera.block.server.exception.BlockStreamProtocolException;
+import com.hedera.block.server.mediator.SubscriptionHandler;
+import com.hedera.block.server.metrics.MetricsService;
+import com.hedera.block.server.notifier.Notifier;
+import com.hedera.block.server.service.ServiceStatus;
+import com.hedera.block.server.verification.service.BlockVerificationService;
+import com.hedera.hapi.block.BlockItemUnparsed;
+import com.hedera.hapi.block.SubscribeStreamResponseUnparsed;
+import com.hedera.pbj.runtime.OneOf;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Verification Handler, receives the block items from the ring buffer, validates their type and uses the BlockVerificationService to verify the block items.
+ */
+@Singleton
+public class StreamVerificationHandlerImpl
+        implements BlockNodeEventHandler<ObjectEvent<SubscribeStreamResponseUnparsed>> {
+
+    private final System.Logger LOGGER = System.getLogger(getClass().getName());
+    private final SubscriptionHandler<SubscribeStreamResponseUnparsed> subscriptionHandler;
+    private final Notifier notifier;
+    private final MetricsService metricsService;
+    private final ServiceStatus serviceStatus;
+    private final BlockVerificationService blockVerificationService;
+
+    private static final String PROTOCOL_VIOLATION_MESSAGE =
+            "Protocol Violation. %s is OneOf type %s but %s is null.\n%s";
+
+    /**
+     * Constructs a new instance of {@link StreamVerificationHandlerImpl}.
+     *
+     * @param subscriptionHandler     the subscription handler
+     * @param notifier                the notifier
+     * @param metricsService          the metrics service
+     * @param serviceStatus           the service status
+     * @param blockVerificationService the block verification service
+     */
+    @Inject
+    public StreamVerificationHandlerImpl(
+            @NonNull final SubscriptionHandler<SubscribeStreamResponseUnparsed> subscriptionHandler,
+            @NonNull final Notifier notifier,
+            @NonNull final MetricsService metricsService,
+            @NonNull final ServiceStatus serviceStatus,
+            @NonNull final BlockVerificationService blockVerificationService) {
+
+        this.subscriptionHandler = subscriptionHandler;
+        this.notifier = notifier;
+        this.metricsService = metricsService;
+        this.serviceStatus = serviceStatus;
+        this.blockVerificationService = blockVerificationService;
+    }
+
+    /**
+     * Handles the event from the ring buffer, unpacks it and uses the BlockVerificationService to verify the block items.
+     */
+    @Override
+    public void onEvent(ObjectEvent<SubscribeStreamResponseUnparsed> event, long l, boolean b) {
+
+        try {
+
+            if (!serviceStatus.isRunning()) {
+                LOGGER.log(ERROR, "Service is not running. Block item will not be processed further.");
```

Comment:
Maybe wrap with else statement, otherwise won't it continue running down ?

Reply:
Thank you for finding this one out, it should have not continue, instead of wrapping with else, added a `return`

✅  Done